b2sdk/sync/action.py

```diff
@@ -389,93 +431,87 @@ def __str__(self):
         """
         Return file size.
 
         :return: always zero
-        :rtype: int
         """
         return 0
 
-    def do_action(self, bucket, reporter):
+    def do_action(self, bucket: Bucket, reporter: ProgressReport):
         """
         Perform the deleting action, returning only after the action is completed.
 
         :param bucket: a Bucket object
-        :type bucket: b2sdk.bucket.Bucket
         :param reporter: a place to report errors
         """
         bucket.api.delete_file_version(self.file_id, self.b2_file_name)
 
-    def do_report(self, bucket, reporter):
+    # TODO: This function uses SyncReport.update_transfer, while others are enough with ProgressReport interface.
+    def do_report(self, bucket: Bucket, reporter: SyncReport):
         """
         Report the deleting action performed.
 
         :param bucket: a Bucket object
-        :type bucket: b2sdk.bucket.Bucket
         :param reporter: a place to report errors
         """
         reporter.update_transfer(1, 0)
         reporter.print_completion('delete ' + self.relative_name + ' ' + self.note)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return 'b2_delete(%s, %s, %s)' % (self.b2_file_name, self.file_id, self.note)
 
 
 class LocalDeleteAction(AbstractAction):
-    def __init__(self, relative_name, full_path):
+    def __init__(self, relative_name: str, full_path: str):
         """
         :param relative_name: a relative file name
-        :type relative_name: str
         :param full_path: a full local path
-        :type: str
         """
         self.relative_name = relative_name
         self.full_path = full_path
 
-    def get_bytes(self):
+    def get_bytes(self) -> int:
         """
         Return file size.
 
         :return: always zero
-        :rtype: int
         """
         return 0
 
-    def do_action(self, bucket, reporter):
+    def do_action(self, bucket: Bucket, reporter: ProgressReport):
         """
         Perform the deleting of a local file action,
         returning only after the action is completed.
 
         :param bucket: a Bucket object
-        :type bucket: b2sdk.bucket.Bucket
         :param reporter: a place to report errors
         """
         os.unlink(self.full_path)
 
-    def do_report(self, bucket, reporter):
+    # TODO: This function uses SyncReport.update_transfer, while others are enough with ProgressReport interface.
+    def do_report(self, bucket: Bucket, reporter: SyncReport):
         """
         Report the deleting of a local file action performed.
 
         :param bucket: a Bucket object
-        :type bucket: b2sdk.bucket.Bucket
         :param reporter: a place to report errors
         """
         reporter.update_transfer(1, 0)
         reporter.print_completion('delete ' + self.relative_name)
 
-    def __str__(self):
-        return 'local_delete(%s)' % (self.full_path)
+    def __str__(self) -> str:
+        return 'local_delete(%s)' % self.full_path
```

Comment:
```suggestion
        return 'local_delete(%s)' % (self.full_path,)
```
or if the type changes, this will explode